crates/polars-plan/src/logical_plan/hive.rs

```diff
@@ -52,8 +54,9 @@ impl HivePartitions {
                 if value.contains('*') {
                     return None;
                 }
-                let value_path = Path::new(value);
-                if value_path.extension().is_some() {
+
+                // Identify file by index location
+                if index == url.display().to_string().split(sep).count() - 1 {
```

Comment:
Can you hoist `url.display().to_string().split(sep).count() - 1` out of the loop? 

And store `url.diplay().to_string()` in a variable first. That saves us doing a string allocation twice.